src/Core/tests/Unit/Test/ResultMockFactoryTest.php

```diff
@@ -80,4 +80,23 @@ public function testCreateFailling()
 
         $result->resolve();
     }
+
+    public function testCreateFaillingWithAwsCodeType()
+    {
+        $result = ResultMockFactory::createFailing(
+            Result::class,
+            400,
+            'Boom',
+            'ResourceNotFoundException',
+            'com.amazonaws.dynamodb.v20120810'
+        );
+
+        try {
+            $result->resolve();
+            self::fail('ClientException should be thrown');
```

Comment:
We should return in the `catch` clause and move this row after the try-catch. 

Say that `$result->resolve();` throws a `\RuntimeException`, then this test wont fail.